modules/infra/cli/port.c

```diff
@@ -231,7 +189,81 @@ static struct cli_context ctx = {
 	.init = ctx_init,
 };
 
+static cmd_status_t l1_xc_set_peer(
+	struct gr_api_client *c,
+	struct gr_iface *iface,
+	const struct ec_pnode *p,
+	uint64_t *set_attrs
+) {
+	struct gr_iface *peer = iface_from_name(c, arg_str(p, "PEER"));
+	if (peer == NULL) {
+		errno = ENODEV;
+		return CMD_ERROR;
+	}
+
+	*set_attrs |= GR_IFACE_SET_MODE;
+	*set_attrs |= GR_IFACE_SET_DOMAIN;
+	iface->mode = GR_IFACE_MODE_L1_XC;
+	iface->domain_id = peer->id;
+	free(peer);
+	return CMD_SUCCESS;
+}
+
+static int l1_xc_init(struct ec_node *mode) {
+	return ec_node_or_add(
+		mode,
+		with_help(
+			"l1 cross connect",
+			EC_NODE_SEQ(
+				"",
+				ec_node_str("", "xconnect"),
+				with_iface_set_callback(
+					l1_xc_set_peer,
+					with_help(
+						"Peer interface for xconnect",
+						ec_node_dyn(
+							"PEER",
+							complete_iface_names,
+							INT2PTR(GR_IFACE_TYPE_PORT)
```

Comment:
This will insert the l1 mode in all interface types (vlan, bond, ipip). It would be interesting to only have this mode available on ports.